api/renterhost_test.go

```diff
@@ -607,3 +607,98 @@ func TestRenterUploadDownload(t *testing.T) {
 		t.Fatal("all new spending should be reflected in metrics:", diff, newSpent.Sub(spent))
 	}
 }
+
+// TestRenterUploadDelete tests that uploading and deleting parallel does not
+// result in failures or stalling.
+func TestRenterUploadDelete(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	t.Parallel()
+	st, err := createServerTester("TestRenterUploadDelete")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer st.server.Close()
+
+	// Announce the host and start accepting contracts.
+	err = st.announceHost()
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = st.acceptContracts()
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = st.setHostStorage()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Set an allowance for the renter, allowing a contract to be formed.
+	allowanceValues := url.Values{}
+	testFunds := "10000000000000000000000000000" // 10k SC
+	testPeriod := "10"
+	allowanceValues.Set("funds", testFunds)
+	allowanceValues.Set("period", testPeriod)
+	err = st.stdPostAPI("/renter", allowanceValues)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a file.
+	path := filepath.Join(st.dir, "test.dat")
+	err = createRandFile(path, 1024)
```

Comment:
this file needs to be a lot bigger, I don't think it's at risk of race condition when it is this small.

Maybe do 10 MB.